src/validation.cpp

```diff
@@ -3095,6 +3097,7 @@ bool Chainstate::ConnectTip(
             if (state.IsInvalid())
                 InvalidBlockFound(pindexNew, state);
             LogError("%s: ConnectBlock %s failed, %s\n", __func__, pindexNew->GetBlockHash().ToString(), state.ToString());
+            view.Reset();
```

Comment:
this isn't strictly needed, right?

Reply:
We don't actually throw away the ephemeral view anymore. We just reset the state each time. This is a big performance improvement since we don't have to reallocate anything. If we didn't do this we would need an external thread pool as well, since the threads are owned by the `CoinsViewCacheAsync`. I think it's a fair tradeoff for one extra line here.
Perhaps calling it `ephemeral_view` is a misnomer though. I'm not sure what a better name would be right now.

Reply:
Renamed to `m_connect_block_view`.